### DISCUSSION TITLE
Problem with installation of coqui-TTS on Ubuntu

### DISCUSSION THREAD

**Discussion by stalevna** [coqui-ai/TTS#1223] [General Q&A]
I am trying to install coquiTTS on Ubuntu 18.04.5 LTS

uname -a returns the following

4.15.0-55-generic #60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

python 3.7

pip version 21.3

When I run the following command:

pip install -e .[all]

I get the error: 

ERROR: Cannot install TTS because these package versions have conflicting dependencies.

The conflict is caused by:
    tensorflow 2.5.0 depends on typing-extensions~=3.7.4
    torch 1.7.0 depends on typing-extensions
    black 22.1.0 depends on typing-extensions>=3.10.0.0; python_version < "3.10"

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies

Did anyone else run into the same problem? Could you share your knowledge on how to fix this?

**Comment by DeluxeMic** [comment]
I did this on Ubuntu 20.04:
python3 -m venv my_venv
source ./my_venv/bin/activate
sudo make system-deps
sudo make install

**Comment by stalevna** [comment]
I downgraded pip to version 20.2 and upgraded python to version 3.8-dev and it helped, packages installed successfully, but I got 


**astroid 2.5.6 requires typed-ast<1.5,>=1.4.0; implementation_name == "cpython" and python_version < "3.8", but you'll have typed-ast 1.5.2 which is incompatible.
tensorboard 2.8.0 requires setuptools>=41.0.0, but you'll have setuptools 39.0.1 which is incompatible.
tensorflow 2.5.0 requires six~=1.15.0, but you'll have six 1.16.0 which is incompatible.
tensorflow 2.5.0 requires typing-extensions~=3.7.4, but you'll have typing-extensions 4.0.1 which is incompatible.
Successfully installed Babel-2.8.1 Jinja2-3.0.3 MarkupSafe-2.0.1 PySocks-1.7.1 TTS Werkzeug-2.0.3 absl-py-0.15.0 anyascii-0.3.0 appdirs-1.4.4 astroid-2.5.6 astunparse-1.6.3 audioread-2.1.9 beautifulsoup4-4.10.0 black-22.1.0 bokeh-1.4.0 cached-property-1.5.2 cachetools-5.0.0 certifi-2021.10.8 cffi-1.15.0 charset-normalizer-2.0.11 click-8.0.3 coqpit-0.0.14 coverage-6.3.1 cycler-0.11.0 cython-0.29.27 decorator-5.1.1 docopt-0.6.2 filelock-3.4.2 flask-2.0.2 flatbuffers-1.12 fonttools-4.29.1 fsspec-2022.1.0 gast-0.4.0 gdown-4.2.1 google-auth-2.6.0 google-auth-oauthlib-0.4.6 google-pasta-0.2.0 grpcio-1.34.1 gruut-1.2.3 gruut-ipa-0.9.3 gruut-lang-cs-1.2 gruut-lang-de-1.2 gruut-lang-es-1.2 gruut-lang-fr-1.2.1 gruut-lang-it-1.2 gruut-lang-nl-1.2 gruut-lang-pt-1.2 gruut-lang-ru-1.2 gruut-lang-sv-1.2 h5py-3.1.0 idna-3.3 importlib-metadata-4.11.0 inflect-5.4.0 isort-5.10.1 itsdangerous-2.0.1 jieba-0.42.1 joblib-1.1.0 jsonlines-1.2.0 keras-nightly-2.5.0.dev2021032900 keras-preprocessing-1.1.2 kiwisolver-1.3.2 lazy-object-proxy-1.7.1 librosa-0.8.0 llvmlite-0.36.0 markdown-3.3.6 matplotlib-3.5.1 mccabe-0.6.1 mecab-python3-1.0.3 mypy-extensions-0.4.3 nose-1.3.7 num2words-0.5.10 numba-0.53.0 numpy-1.19.5 oauthlib-3.2.0 opt-einsum-3.3.0 packaging-21.3 pandas-1.3.5 pathspec-0.9.0 pillow-9.0.1 platformdirs-2.5.0 pooch-1.6.0 protobuf-3.19.4 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycparser-2.21 pylint-2.8.3 pynndescent-0.5.6 pyparsing-3.0.7 pypinyin-0.45.0 pysbd-0.3.4 python-crfsuite-0.9.7 python-dateutil-2.8.2 pytz-2021.3 pyyaml-6.0 requests-2.27.1 requests-oauthlib-1.3.1 resampy-0.2.2 rsa-4.8 scikit-learn-1.0.2 scipy-1.7.3 six-1.16.0 soundfile-0.10.3.post1 soupsieve-2.3.1 tensorboard-2.8.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 tensorboardX-2.4.1 tensorflow-2.5.0 tensorflow-estimator-2.5.0 termcolor-1.1.0 threadpoolctl-3.1.0 toml-0.10.2 tomli-2.0.1 torch-1.10.2 tornado-6.1 tqdm-4.62.3 typed-ast-1.5.2 typing-extensions-4.0.1 umap-learn-0.5.1 unidic-lite-1.0.8 urllib3-1.26.8 wrapt-1.12.1 zipp-3.7.0
WARNING: You are using pip version 20.2; however, version 22.0.3 is available.**

I changed setuptools to version 41.0.0, six to 1.15 and typing-extensions=3.7.4 and I got a different dependency issue
black 22.1.0 requires typing-extensions 3.10.0

I stopped here and will try to just run it and see if this dependencies affect anything

**Comment by DeluxeMic** [comment]
Like I already answered. Use virtual enviroment to avoid depency problems: apt install python3.8-venv
python3 -m venv my_venv
source ./my_venv/bin/activate
sudo make system-deps
sudo make install

https://www.freecodecamp.org/news/how-to-manage-python-dependencies-using-virtual-environments

**Reply by stalevna** [reply]
I have used venv, I created it specifically for each python version I tried 3.6, 3,7 and 3.8
and makefile runs pip install -e .[all] anyways, so I don't see the difference whether you run the command or make install

If it worked in your system, good for you. I asked people who have experienced such a problem. So no, you haven't answered the question I asked

**Comment by DeluxeMic** [comment]
Did you also: sudo make system-deps? Makefile says that it does: sudo apt-get install -y libsndfile1-dev. I had also same errors with pip only install.
